test.java

```diff
@@ -0,0 +1,34 @@
+import java.util.Scanner;
```

Comment:
*DevCompanion review for test.java:*
Overall, the provided Java code looks secure and follows best practices for secure coding, input validation, and error handling. However, there is one potential CWE issue that needs to be addressed:

CWE-586: Explicit Resource Closure Error
- Line 23: The `scanner.close()` method is called after reading user input. However, since the `Scanner` object is created using `System.in`, closing it will also close the underlying `System.in` stream, which may cause issues if the program needs to read input again later. It is recommended to avoid closing `System.in` and instead close only resources that are explicitly opened by the program.

Suggested correction:
- Remove the line `scanner.close()` to avoid closing `System.in`.

Explanation:
Closing `System.in` can lead to unexpected behavior if the program needs to read input again later. By removing the `scanner.close()` statement, the program ensures that the `System.in` stream remains open and can be used for future input reading if needed.

Changes: Required